plugins/destination/file/client/client.go

```diff
@@ -0,0 +1,112 @@
+package client
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"os"
+	"strings"
+
+	"cloud.google.com/go/storage"
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/config"
+	"github.com/aws/aws-sdk-go-v2/feature/s3/manager"
+	"github.com/aws/aws-sdk-go-v2/service/s3"
+	"github.com/cloudquery/plugin-sdk/plugins"
+	"github.com/cloudquery/plugin-sdk/specs"
+	"github.com/rs/zerolog"
+)
+
+type Client struct {
+	plugins.DefaultReverseTransformer
+	logger  zerolog.Logger
+	spec    specs.Destination
+	csvSpec Spec
+
+	baseDir string
+	path    string
+
+	awsUploader   *manager.Uploader
+	awsDownloader *manager.Downloader
+
+	gcpStorageClient *storage.Client
+	gcpBucket        *storage.BucketHandle
+
+	// testMode bool
+
+	metrics plugins.DestinationMetrics
+}
+
+func New(ctx context.Context, logger zerolog.Logger, spec specs.Destination) (plugins.DestinationClient, error) {
+	if spec.WriteMode != specs.WriteModeAppend {
+		return nil, fmt.Errorf("file destination only supports append mode")
+	}
+	c := &Client{
+		logger: logger.With().Str("module", "dest-file").Logger(),
+		spec:   spec,
+	}
+
+	if err := spec.UnmarshalSpec(&c.csvSpec); err != nil {
+		return nil, fmt.Errorf("failed to unmarshal postgresql spec: %w", err)
+	}
+	c.csvSpec.SetDefaults()
+	split := strings.Split(c.csvSpec.Directory, "/")
+	if len(split) == 0 {
+		return nil, fmt.Errorf("invalid directory: %s", c.csvSpec.Directory)
+	}
+	c.baseDir = split[0]
+	if len(split) > 1 {
+		c.path = strings.Join(split[1:], "/")
+	} else {
+		c.path = ""
+	}
+
+	switch c.csvSpec.Backend {
+	case BackendTypeLocal:
+		if err := os.MkdirAll(c.csvSpec.Directory, 0755); err != nil {
+			return nil, fmt.Errorf("failed to create directory: %w", err)
+		}
+	case BackendTypeS3:
+		awsCfg, err := config.LoadDefaultConfig(ctx)
+		if err != nil {
+			return nil, fmt.Errorf("unable to load AWS SDK config, %w", err)
```

Comment:
```suggestion
			return nil, fmt.Errorf("unable to load AWS SDK config: %w", err)
```
err conversion is `wrapper text: %w`